skill-tree/src/main/java/com/RDS/skilltree/utils/JWTAuthenticationFilter.java

```diff
@@ -48,6 +49,12 @@ public void doFilterInternal(
         filterChain.doFilter(request, response);
     }
 
+    @Override
```

Comment:
in this method iterates through the entire list of permitted paths using stream().anyMatch(). This could be inefficient for large lists. Consider using a Set data structure for PERMITTED_PATHS instead of a List. Sets offer faster lookups using the contains method, improving performance for larger lists.

```
@Override
protected boolean shouldNotFilter(HttpServletRequest request) throws ServletException {
  String path = request.getRequestURI();
  return   
 SecurityConfig.PERMITTED_PATHS.contains(path);
}
```

Reply:
- the `PERMITTED_ROUTES` list will generally not be large as it only stores the list of public routes.